File: PDFPropertyEdit/PDFPropertyEdit/XmlSettings.vb
Imports System
Imports System.IO
Imports System.Text
Imports System.Runtime.Serialization
Imports System.Runtime.Serialization.Formatters.Binary
Imports System.Windows.Forms
Imports Microsoft.Win32

<Serializable()>
<System.Diagnostics.DebuggerStepThrough()>
Public Class XmlSettings

#Region "プライベート変数"

	'設定を保存するフィールド
	'フォーム全般
	Private _LocationX As Integer   'フォーム位置X
	Private _LocationY As Integer   'フォーム位置Y
	Private _SizeX As Integer   'フォーム幅
	Private _SizeY As Integer   'フォーム高さ
	Private _State As Integer   '最大化・最小化状態
	Private _ApplicationName As String  'アプリケーション名

	Private _TargetPath As String
	Private _OutPath As String
	Private _Title As String
	Private _Author As String
	Private _SubTitle As String
	Private _Keywords As String
	Private _CSVFile As String
	Private _TIFTargetPath As String
	Private _TIFSubject As String
	Private _TIFAuthor As String

	'SQLServer接続用
	Private _DataSource As String
	Private _InitialCatalog As String
	Private _UserID As String
	Private _Password As String
	Private _TablePrefix As String

#End Region

#Region "プロパティ"

	''' <summary>
	''' フォーム位置X
	''' </summary>
	''' <returns></returns>
	Public Property LocationX() As Integer
		Get
			Return _LocationX
		End Get
		Set(value As Integer)
			_LocationX = value
		End Set
	End Property

	''' <summary>
	''' フォーム位置Y
	''' </summary>
	''' <returns></returns>
	Public Property LocationY() As Integer
		Get
			Return _LocationY
		End Get
		Set(value As Integer)
			_LocationY = value
		End Set
	End Property

	''' <summary>
	''' サイズ幅
	''' </summary>
	''' <returns></returns>
	Public Property SizeX() As Integer
		Get
			Return _SizeX
		End Get
		Set(value As Integer)
			_SizeX = value
		End Set
	End Property

	''' <summary>
	''' サイズ高さ
	''' </summary>
	''' <returns></returns>
	Public Property SizeY() As Integer
		Get
			Return _SizeY
		End Get
		Set(value As Integer)
			_SizeY = value
		End Set
	End Property

	''' <summary>
	''' 最大化・最小化状態
	''' </summary>
	''' <returns></returns>
	Public Property State() As Integer
		Get
			Return _State
		End Get
		Set(value As Integer)
			_State = value
		End Set
	End Property

	''' <summary>
	''' アプリケーション名
	''' </summary>
	''' <returns></returns>
	Public Property ApplicationName() As String
		Get
			Return _ApplicationName
		End Get
		Set(value As String)
			_ApplicationName = value
		End Set
	End Property

	''' <summary>
	''' DataSource
	''' </summary>
	''' <value></value>
	''' <returns></returns>
	''' <remarks></remarks>
	Public Property DataSource() As String
		Get
			Return _DataSource
		End Get
		Set(value As String)
			_DataSource = value
		End Set
	End Property

	''' <summary>
	''' InitialCatalog
	''' </summary>
	''' <value></value>
	''' <returns></returns>
	''' <remarks></remarks>
	Public Property InitialCatalog() As String
		Get
			Return _InitialCatalog
		End Get
		Set(value As String)
			_InitialCatalog = value
		End Set
	End Property

	''' <summary>
	''' UserID
	''' </summary>
	''' <value></value>
	''' <returns></returns>
	''' <remarks></remarks>
	Public Property UserID() As String
		Get
			Return _UserID
		End Get
		Set(value As String)
			_UserID = value
		End Set
	End Property

	''' <summary>
	''' Password
	''' </summary>
	''' <value></value>
	''' <returns></returns>
	''' <remarks></remarks>
	Public Property Password() As String
		Get
			Return _Password
		End Get
		Set(value As String)
			_Password = value
		End Set
	End Property

	''' <summary>
	''' INTRA-PDC00接続時の接頭語入出力
	''' </summary>
	''' <returns></returns>
	Public Property TablePrefix() As String
		Get
			Return _TablePrefix
		End Get
		Set(value As String)
			_TablePrefix = value
		End Set
	End Property

	''' <summary>
	''' PDF対象フォルダ
	''' </summary>
	''' <returns></returns>
	Public Property TargetPath() As String
		Get
			Return _TargetPath
		End Get
		Set(value As String)
			_TargetPath = value
		End Set
	End Property

	''' <summary>
	''' PDF出力フォルダ
	''' </summary>
	''' <returns></returns>
	Public Property OutPath() As String
		Get
			Return _OutPath
		End Get
		Set(value As String)
			_OutPath = value
		End Set
	End Property

	''' <summary>
	''' PDFタイトル
	''' </summary>
	''' <returns></returns>
	Public Property Title() As String
		Get
			Return _Title
		End Get
		Set(value As String)
			_Title = value
		End Set
	End Property

	''' <summary>
	''' PDF作成者
	''' </summary>
	''' <returns></returns>
	Public Property Author() As String
		Get
			Return _Author
		End Get
		Set(value As String)
			_Author = value
		End Set
	End Property

	''' <summary>
	''' サブタイトル
	''' </summary>
	''' <returns></returns>
	Public Property SubTitle() As String
		Get
			Return _SubTitle
		End Get
		Set(value As String)
			_SubTitle = value
		End Set
	End Property

	''' <summary>
	''' PDFキーワード
	''' </summary>
	''' <returns></returns>
	Public Property Keywords() As String
		Get
			Return _Keywords
		End Get
		Set(value As String)
			_Keywords = value
		End Set
	End Property

	''' <summary>
	''' CSVファイル
	''' </summary>
	''' <returns></returns>
	Public Property CSVFile() As String
		Get
			Return _CSVFile
		End Get
		Set(value As String)
			_CSVFile = value
		End Set
	End Property

	''' <summary>
	''' TIF対象フォルダ
	''' </summary>
	''' <returns></returns>
	Public Property TIFTargetPath() As String
		Get
			Return _TIFTargetPath
		End Get
		Set(value As String)
			_TIFTargetPath = value
		End Set
	End Property

	''' <summary>
	''' 件名
	''' </summary>
	''' <returns></returns>
	Public Property TIFSubject() As String
		Get
			Return _TIFSubject
		End Get
		Set(value As String)
			_TIFSubject = value
		End Set
	End Property

	''' <summary>
	''' TIF作成者
	''' </summary>
	''' <returns></returns>
	Public Property TIFAuthor() As String
		Get
			Return _TIFAuthor
		End Get
		Set(value As String)
			_TIFAuthor = value
		End Set
	End Property

#End Region

#Region "コンストラクタ"

	Public Sub New()

		'フォーム全般
		_LocationX = 0
		_LocationY = 0
		_SizeX = 1024
		_SizeY = 768
		_ApplicationName = "JP健診 就業判定票発送業務支援ツール"

		_TargetPath = ""
		_OutPath = ""
		_Title = ""
		_Author = ""
		_SubTitle = ""
		_Keywords = ""
		_CSVFile = ""
		_TIFTargetPath = ""
		_TIFSubject = "国民所得推計研究会資料"
		_TIFAuthor = "一橋大学経済研究所附属社会科学統計情報研究センター"

		'SQLServer接続用
		_DataSource = "JPSERV\INTRASQL"
		_InitialCatalog = "JPHealth"
		_UserID = "sa"
		_Password = "intra_intra"
		_TablePrefix = "[DailyReport_Test].[dbo]."

	End Sub

#End Region

#Region "インスタンス"

	'XmlSettingsクラスのただ一つのインスタンス
	<NonSerialized()>
	Private Shared _instance As XmlSettings

	<System.Xml.Serialization.XmlIgnore()>
	Public Shared Property Instance() As XmlSettings
		Get
			If _instance Is Nothing Then
				_instance = New XmlSettings
			End If
			Return _instance
		End Get
		Set(value As XmlSettings)
			_instance = value
		End Set
	End Property

#End Region

#Region "設定の読み書き"

	''' <summary>
	''' 設定をXMLファイルから読み込み復元する
	''' </summary>
	''' <remarks></remarks>
	Public Shared Sub LoadFromXmlFile()

		Dim p As String = GetSettingPath()
		If Not System.IO.File.Exists(p) Then
			Exit Sub
		End If
		Dim sr As New StreamReader(p, New UTF8Encoding(False))
		Dim xs As New System.Xml.Serialization.XmlSerializer(GetType(XmlSettings))
		'読み込んで逆シリアル化する
		Dim obj As Object = xs.Deserialize(sr)
		sr.Close()

		Instance = CType(obj, XmlSettings)

	End Sub

	''' <summary>
	''' 現在の設定をXMLファイルに保存する
	''' </summary>
	''' <remarks></remarks>
	Public Shared Sub SaveToXmlFile()

		Dim p As String = GetSettingPath()
		If Not System.IO.Directory.Exists(System.IO.Path.GetDirectoryName(p)) Then
			My.Computer.FileSystem.CreateDirectory(System.IO.Path.GetDirectoryName(p))
		End If
		Dim sw As New StreamWriter(p, False, New UTF8Encoding(False))
		Dim xs As New System.Xml.Serialization.XmlSerializer(GetType(XmlSettings))
		'シリアル化して書き込む
		xs.Serialize(sw, Instance)
		sw.Close()

	End Sub


	''' <summary>
	''' 設定をバイナリファイルから読み込み復元する
	''' </summary>
	Public Shared Sub LoadFromBinaryFile()
		Dim p As String = GetSettingPath()

		Dim fs As New FileStream(p, FileMode.Open, FileAccess.Read)
		Dim bf As New BinaryFormatter
		'読み込んで逆シリアル化する
		Dim obj As Object = bf.Deserialize(fs)
		fs.Close()

		Instance = CType(obj, XmlSettings)
	End Sub

	''' <summary>
	''' 現在の設定をバイナリファイルに保存する
	''' </summary>
	Public Shared Sub SaveToBinaryFile()
		Dim p As String = GetSettingPath()

		Dim fs As New FileStream(p, FileMode.Create, FileAccess.Write)
		Dim bf As New BinaryFormatter
		'シリアル化して書き込む
		bf.Serialize(fs, Instance)
		fs.Close()
	End Sub

	''' <summary>
	''' 設定をレジストリから読み込み復元する
	''' </summary>
	Public Shared Sub LoadFromRegistry()
		Dim bf As New BinaryFormatter
		'レジストリから読み込む
		Dim reg As RegistryKey = GetSettingRegistry()
		Dim bs As Byte() = CType(reg.GetValue(""), Byte())
		'逆シリアル化して復元
		Dim ms As New MemoryStream(bs, False)
		Instance = CType(bf.Deserialize(ms), XmlSettings)
		'閉じる
		ms.Close()
		reg.Close()
	End Sub

	''' <summary>
	''' 現在の設定をレジストリに保存する
	''' </summary>
	Public Shared Sub SaveToRegistry()
		Dim ms As New MemoryStream
		Dim bf As New BinaryFormatter
		'シリアル化してMemoryStreamに書き込む
		bf.Serialize(ms, Instance)
		'レジストリへ保存する
		Dim reg As RegistryKey = GetSettingRegistry()
		reg.SetValue("", ms.ToArray())
		'閉じる
		ms.Close()
		reg.Close()
	End Sub

	''' <summary>
	''' インスタンスの内容を保存して解放する
	''' </summary>
	''' <remarks></remarks>
	Public Shared Sub Dispose()

		SaveToXmlFile()
		Instance = Nothing

	End Sub

#End Region

#Region "プライベートメソッド"

	''' <summary>
	''' XMLファイルの保存先を取得する
	''' </summary>
	''' <returns></returns>
	''' <remarks></remarks>
	Private Shared Function GetSettingPath() As String

		'Dim p As String = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData),
		'							   Application.CompanyName + "\" + Application.ProductName _
		'							   + "\" + "Settings.xml")
		Dim p As String = Application.StartupPath & "\" + "Settings.xml"

		Return p

	End Function

	''' <summary>
	''' レジストリの保存先キーを取得する
	''' </summary>
	''' <returns></returns>
	''' <remarks></remarks>
	Private Shared Function GetSettingRegistry() As RegistryKey

		Dim reg As RegistryKey = Registry.CurrentUser.CreateSubKey(("Software\" + Application.CompanyName +
																	"\" + Application.ProductName))
		Return reg

	End Function

#End Region

End Class

File: PDFしおりリンクツール/PDF_Shiori/Module/XmlSettings.cs
using System;
using System.IO;
using System.Runtime.Serialization;
using System.Runtime.Serialization.Formatters.Binary;
using System.Windows.Forms;
using Microsoft.Win32;
using System.Diagnostics;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PDF_Shiori
{
	[Serializable]
	[DebuggerStepThrough]
	public class XmlSettings
	{
		#region プライベート変数
		//設定を保存するフィールド
		//フォーム全般
		private static int _LocationX;
		private static int _LocationY;
		private static int _SizeX;
		private static int _SizeY;
		private static int _State;
		private static string _ApplicationName;
		//SQL Server関連
		private static string _DataSource;
		private static string _InitialCatalog;
		private static string _UserID;
		private static string _Password;

		#endregion

		#region プロパティ

		/*==============================*/
		//フォーム全般
		/*==============================*/

		/// <summary>
		/// フォーム位置X
		/// </summary>
		public int LocationX
		{
			get
			{
				return _LocationX;
			}
			set
			{
				_LocationX = value;
			}
		}

		/// <summary>
		/// フォーム位置Y
		/// </summary>
		public int LocationY
		{
			get
			{
				return _LocationY;
			}
			set
			{
				_LocationY = value;	
			}
		}

		/// <summary>
		/// サイズ幅
		/// </summary>
		public int SizeX
		{
			get
			{
				return _SizeX;
			}
			set
			{
				_SizeX = value;	
			}
		}

		/// <summary>
		/// サイズ高さ
		/// </summary>
		public int SizeY
		{
			get
			{
				return _SizeY;
			}
			set
			{
				_SizeY = value;	
			}
		}

		/// <summary>
		/// 最大化・最小化状態
		/// </summary>
		public int State
		{
			get
			{
				return _State;
			}
			set
			{
				_State = value;
			}
		}

		/// <summary>
		/// アプリケーション名
		/// </summary>
		public string ApplicationName
		{
			get
			{
				return _ApplicationName;
			}
			set
			{
				_ApplicationName = value;
			}
		}

		/*==============================*/
		//SQL Server関連
		/*==============================*/

		/// <summary>
		/// DataSource
		/// </summary>
		public string DataSource
		{
			get
			{
				return _DataSource;
			}
			set
			{
				_DataSource = value;	
			}
		}

		/// <summary>
		/// InitialCatalog
		/// </summary>
		public string InirialCatalog
		{
			get
			{
				return _InitialCatalog;
			}
			set
			{
				_InitialCatalog = value;	
			}
		}

		/// <summary>
		/// UserID
		/// </summary>
		public string UserID
		{
			get
			{
				return _UserID;
			}
			set
			{
				_UserID = value;	
			}
		}

		/// <summary>
		/// Password
		/// </summary>
		public string Password
		{
			get
			{
				return _Password;
			}
			set
			{
				_Password = value;
			}
		}
		#endregion

		#region コンストラクタ

		public XmlSettings()
		{
			//フォーム全般
			_LocationX = 0;
			_LocationY = 0;
			_SizeX = 1024;
			_SizeY = 768;
			_ApplicationName = "なんちゃらツール";
			//SQL Server関連
			_DataSource = "INTRA-PDC00\\INTRASQL";
			_InitialCatalog = "InTools";
			_UserID = "sa";
			_Password = "intra_intra";
		}
		#endregion

		#region インスタンス

		//XmlSettingsクラスのただひとつのインスタンス
		[NonSerialized]
		private static XmlSettings _instance;

		[System.Xml.Serialization.XmlIgnore]
		public static XmlSettings Instance
		{
			get
			{
				if (_instance == null)
				{
					_instance = new XmlSettings();
				}
				return _instance;
			}
			set
			{
				_instance = value;
			}
		}

		#endregion

		#region 設定の読み書き

		/// <summary>
		/// 設定をXMLファイルから読み込み復元する
		/// </summary>
		public static void LoadFromXmlFile()
		{
			string p = GetSettingPath();
			if (File.Exists(p) == false)
			{
				return;
			}
			StreamReader sr = new StreamReader(p, new UTF8Encoding(false));
			System.Xml.Serialization.XmlSerializer xs = new System.Xml.Serialization.XmlSerializer(typeof(XmlSettings));
			//読み込んでシリアル化する
			object obj = xs.Deserialize(sr);
			sr.Close();

			Instance = (XmlSettings)obj;
		}

		/// <summary>
		/// 現在の設定をXMLファイルに保存する
		/// </summary>
		public static void SaveToXmlFile()
		{
			string p = GetSettingPath();
			if (Directory.Exists(Path.GetDirectoryName(p)) == false)
			{
				DirectoryInfo di = Directory.CreateDirectory(System.IO.Path.GetDirectoryName(p));
			}
			StreamWriter sw = new StreamWriter(p, false, new UTF8Encoding(false));
			System.Xml.Serialization.XmlSerializer xs = new System.Xml.Serialization.XmlSerializer(typeof(XmlSettings));
			//シリアル化して書き込む
			xs.Serialize(sw, Instance);
			sw.Close();
		}

		/// <summary>
		/// インスタンスの内容を保存して解放する
		/// </summary>
		public static void Dispose()
		{
			SaveToXmlFile();
			Instance = null;
		}
		#endregion

		#region プライベートメソッド
		
		/// <summary>
		/// XMLファイルの保存先を取得する
		/// </summary>
		/// <returns></returns>
		private static string GetSettingPath()
		{
			string p = Application.StartupPath + "\\Settings.xml";
			return p;
		}
		#endregion
	}
}
